dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/DefaultTrackedEntityAttributeService.java

```diff
@@ -303,10 +303,17 @@ public Set<TrackedEntityAttribute> getAllUserReadableTrackedEntityAttributes(
 
   @Override
   @Transactional(readOnly = true)
-  public Set<TrackedEntityAttribute> getAllUserReadableTrackedEntityAttributes(Program program) {
-    List<TrackedEntityType> trackedEntityTypes = trackedEntityTypeService.getAllTrackedEntityType();
+  public Set<TrackedEntityAttribute> getProgramAttributes(Program program) {
+    return getAllUserReadableTrackedEntityAttributes(
+        CurrentUserUtil.getCurrentUserDetails(), List.of(program), List.of());
+  }
+
+  @Override
+  @Transactional(readOnly = true)
+  public Set<TrackedEntityAttribute> getTrackedEntityTypeAttributes(
+      TrackedEntityType trackedEntityType) {
     return getAllUserReadableTrackedEntityAttributes(
-        CurrentUserUtil.getCurrentUserDetails(), List.of(program), trackedEntityTypes);
+        CurrentUserUtil.getCurrentUserDetails(), List.of(), List.of(trackedEntityType));
   }
```

Comment:
If this runs with no security context the user can be null, but maybe it is not the case here

Reply:
At least for now its only run in sync as its only used in the GET of file/images.

I am going to bring this up in the backend meeting though. I think `CurrentUserUtil.getCurrentUserDetails()` should throw a checked exception as most of the time you expect the user to be there and anything else is an exception. We could then add a method that is used in the rare cases where you want to deal with the user not being set.